src/Integration.Vsix/SonarLintDaemon/SonarLintDaemon.cs

```diff
@@ -242,28 +242,42 @@ private void WritelnToPane(string msg)
             logger.WriteLine(msg);
         }
 
-        // Need to be able to stub this method out for testing 
-        public virtual /* for testing */ void Stop()
+        public void Stop()
         {
             if (!IsRunning)
             {
                 return;
                 // throw exception?
             }
 
+            this.SafeInternalStop();
+        }
+
+        // Need to be able to stub this method out for testing 
+        protected virtual void SafeInternalStop()
+        {
+            // Note: we're not checking IsRunning here as that check can fail if the
+            // process wasn't started correctly.
             logger.WriteLine(Strings.Daemon_Stopping);
-            daemonClient = null;
-            channel?.ShutdownAsync().Wait();
-            channel = null;
-            process?.Kill();
-            process?.WaitForExit();
-            process = null;
+
+            SafeOperation(() =>
+            {
+                daemonClient = null;
+                channel?.ShutdownAsync().Wait();
+                channel = null;
+
+                // Will throw an InvalidOperationException if the process isn't valid
+                process?.Kill();
+                process?.WaitForExit();
+                process = null;
+            });
+
             logger.WriteLine(Strings.Daemon_Stopped);
         }
 
         public int Port { get; private set; }
 
-        public bool IsInstalled => Directory.Exists(InstallationPath) && File.Exists(ExePath);
+        public bool IsInstalled => File.Exists(ExePath);
```

Comment:
Removed the redundant directory check (the exe is always under the installation path).